modules/ui/src/main/scala/lucuma/ui/forms/FormInputEV.scala

```diff
@@ -77,36 +81,73 @@ object FormInputEV {
   type Scope[EV[_], A]   = RenderScope[Props[EV, A], State, Unit]
 
   @Lenses
-  final case class State(curValue: String, prevValue: String)
-
-  def onTextChange[EV[_], A]($ : Scope[EV, A]): ReactEventFromInput => Callback =
-    (e: ReactEventFromInput) => {
-      // Capture the value outside setState, react reuses the events
-      val v = e.target.value
-      // First update the internal state, then call the outside listener
-      $.setStateL(State.curValue)(v) *>
-        // Next 2 might not be called if the InputFormat returns None
-        $.props.valSet(v) *>
-        $.props.onChangeC(v)
-    }
-
-  def onBlur[EV[_], A]($ : Scope[EV, A], c: ChangeCallback[String]): Callback =
-    c($.state.curValue)
+  final case class State(displayValue: String, modelValue: String, cursor: Option[(Int, Int)])
 
-  protected val component =
-    ScalaComponent
-      .builder[Props[Any, Any]]
-      .getDerivedStateFromPropsAndState[State] { (props, stateOpt) =>
-        val newValue = props.valGet
-        // Force new value from props if the prop changes (or we are initializing).
-        stateOpt match {
-          case Some(state) if newValue === state.prevValue => state
-          case _                                           => State(newValue, newValue)
+  class Backend[EV[_], A]($ : BackendScope[Props[EV, A], State]) {
+    private val outerRef = Ref[html.Element]
+
+    def getInputElement(e: html.Element): html.Input =
```

Comment:
This feels a bit fragile as it depends on the structure of the dom. Could we do a search from `e` via e.g. a cclass name?. I guess it would still be fragile but would survive some structure changes.

Reply:
Yeah, it's definitely fragile... Searching by class name is a good suggestion. I'll see if I can make that work.

Reply:
Unfortunately, there are no classes on the input itself. :( The best we could do would be to search for the `input` class on the div that contains the input. I guess that would be an improvement in safety, but would add another layer of nondeterminism in having to deal with the search failing.